server/service/http/message/query/query_structure.rs

```diff
@@ -0,0 +1,227 @@
+            let role_type_2 = query_structure_role_type_as_vertex(query_structure, indexed.role_type_2());
+            push_edge!(edges, query_structure, indexed.relation(), indexed.player_1(), Links(role_type_1));
+            push_edge!(edges, query_structure, indexed.relation(), indexed.player_2(), Links(role_type_2));
+        }
+        Constraint::ExpressionBinding(expr) => {
+            let expr_vertex =
+                { EncodedQueryStructureVertex::Expression { repr: format!("Expression#{index}").to_owned() } }; // TODO
+            expr.ids_assigned().for_each(|variable| {
+                let assigned = query_structure_vertex(query_structure, &Vertex::Variable(variable));
+                let assigned_name =
+                    query_structure.get_variable_name(&variable).unwrap_or_else(|| variable.to_string());
+                let edge_type = EncodedQueryStructureEdgeType::Assigned(assigned_name);
+                edges.push(EncodedQueryStructureEdge { r#type: edge_type, from: expr_vertex.clone(), to: assigned });
+            });
+            expr.required_ids().for_each(|variable| {
+                let argument = query_structure_vertex(query_structure, &Vertex::Variable(variable));
+                let arg_name = query_structure.get_variable_name(&variable).unwrap_or_else(|| variable.to_string());
+                let edge_type = EncodedQueryStructureEdgeType::Argument(arg_name);
+                edges.push(EncodedQueryStructureEdge { r#type: edge_type, from: argument, to: expr_vertex.clone() });
+            });
+        }
+        Constraint::FunctionCallBinding(function_call) => {
+            let func_vertex =
+                { EncodedQueryStructureVertex::FunctionCall { repr: format!("Function#{index}").to_owned() } }; // TODO
+            function_call.ids_assigned().for_each(|variable| {
+                let assigned = query_structure_vertex(query_structure, &Vertex::Variable(variable));
+                let assigned_name =
+                    query_structure.get_variable_name(&variable).unwrap_or_else(|| variable.to_string());
+                let edge_type = EncodedQueryStructureEdgeType::Assigned(assigned_name);
+                edges.push(EncodedQueryStructureEdge { r#type: edge_type, from: func_vertex.clone(), to: assigned });
+            });
+            function_call.required_ids().for_each(|variable| {
+                let argument = query_structure_vertex(query_structure, &Vertex::Variable(variable));
+                let arg_name = query_structure.get_variable_name(&variable).unwrap_or_else(|| variable.to_string());
+                let edge_type = EncodedQueryStructureEdgeType::Argument(arg_name);
+                edges.push(EncodedQueryStructureEdge { r#type: edge_type, from: argument, to: func_vertex.clone() });
+            });
+        }
+        | Constraint::Comparison(_) => {
+            // unimplemented_feature!(GraphViz);
```

Comment:
We haven't decided how to represent this basically right?

Reply:
Yeah. Or we can just not.

Reply:
happy to leave this as not for the time being for sure